### PR TITLE
feat: add support for flexible checksums on Normal payloads

### PR DESCRIPTION
**Pull request by dayaffe**
## Issue \#
#1287 

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sichanyoo — APPROVED** [review]
Lgtm

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    override fun enabledForService(model: Model, settings: SwiftSettings): Boolean = model
+        .shapes<OperationShape>()
+        .any { it.hasTrait(HttpChecksumTrait::class.java) }
+
+    override fun customizeMiddleware(
+        ctx: ProtocolGenerator.GenerationContext,
+        operationShape: OperationShape,
+        operationMiddleware: OperationMiddleware,
+    ) {
+        val httpChecksumTrait = operationShape.getTrait(HttpChecksumTrait::class.java).getOrNull()
```
Comment (jbelkins) [inline codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt:31]: Is this the same `.getOrNull()` that I'm eliminating in the XML branch?

**Review by jbelkins — COMMENTED** [review]
Just a question about `getOrNull()`

File: codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt (hunk 1 above)
Comment (jbelkins) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt:31]: (I think we decided to use `.orElse(null)` which is a Kotlin built-in)

Comment (dayaffe) [reply codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt:31]: the `.orElse(null)` we were talking about I believe works on Java types not Kotlin types where we prefer to use `?.` that work on optionals. I can check where .getOrNull() is used and see if we can switch it over to the kotlin optionals

**Review by jbelkins — APPROVED** [review]

**Pull request merged**